joplin/pages/event_page/factories.py

```diff
@@ -0,0 +1,72 @@
+import factory
+from pages.base_page.factories import JanisBasePageFactory
+from pages.home_page.models import HomePage
+from pages.event_page.models import EventPage
+
+class EventPageFactory(JanisBasePageFactory):
+    class Meta:
+        model = EventPage
+
+def create_event_page_from_importer_dictionaries(page_dictionaries, revision_id):
+    # first check to see if we already imported this page
+    # if we did, just go to the edit page for it without changing the db
+    # todo: maybe change this to allow updating pages in the future?
+    try:
+        page = EventPage.objects.get(imported_revision_id=revision_id)
+    except EventPage.DoesNotExist:
+        page = None
+    if page:
+        return page
+
+    # since we don't have a page matching the revision id, we should look
+    # for other matches, for now let's just use english slug
+    # todo: figure out what we want the logic around importing a page with the same slug to look like
+    try:
+        page = EventPage.objects.get(slug=page_dictionaries['en']['slug'])
+    except EventPage.DoesNotExist:
+        page = None
+    if page:
+        return page
+
+    # since we don't have a page matching the revision id or the slug
+    # make the combined page dictionary
+    combined_dictionary = page_dictionaries['en']
+
+    # todo: figure out where to move this so it isn't copypasta in service page
```

Comment:
just a little note to clean this up

Reply:
Ah! Sorry, should have left a note here... 
- I wasn't sure what we all need or want for the future. I just had to add some of it to get it working. But, not sure what is/will be needed or not.